test/ContractTestContext.ts

```diff
@@ -90,107 +86,99 @@ export type ContractTestContext = ArchContracts & {
+    await context.lvUSDToken.approve(context.coordinator.address, ethers.constants.MaxUint256);
 
-    await context.lvUSD.approve(context.exchanger.address, ethers.constants.MaxUint256);
-    await context.lvUSD.approve(context.coordinator.address, ethers.constants.MaxUint256);
+    /* expected contract roles when deployed to mainnet: */
+    const contractRoles = {
+        positionToken: {
+            executive: context.leverageEngine.address,
+        },
+    };
 
-    // Post init contracts
-    await Promise.all([
-        context.leverageEngine.init(
+    const initArgs = {
+        leverageEngine: [
             context.coordinator.address,
             context.positionToken.address,
             context.parameterStore.address,
             context.leverageAllocator.address,
             context.externalOUSD.address,
-        ),
-
-        context.coordinator.init(
-            context.lvUSD.address,
-            context.vault.address,
-            context.cdp.address,
+        ],
+        coordinator: [
+            context.lvUSDToken.address,
+            context.vaultOUSD.address,
+            context.cDPosition.address,
             context.externalOUSD.address,
             context.exchanger.address,
             context.parameterStore.address,
-        ),
-
-        context.exchanger.init(
+        ],
+        exchanger: [
             context.parameterStore.address,
             context.coordinator.address,
-            context.lvUSD.address,
+            context.lvUSDToken.address,
             context.externalOUSD.address,
             context.external3CRV.address,
             context.curveLvUSDPool.address,
             addressCurveOUSDPool,
-        ),
+        ],
+        vaultOUSD: [context.parameterStore.address, context.externalOUSD.address],
+        parameterStore: [context.treasurySigner.address],
+    };
 
-        context.vault.init(context.parameterStore.address, context.externalOUSD.address),
-        context.parameterStore.init(context.treasurySigner.address),
-        context.positionToken.init(),
-    ]);
+    /* call setRoles and init on all contracts, allowing any specified overrides from arguments: */
+    await Promise.all(Object.entries(contracts).map(async ([contractKey, contract]) => {
+        const roles = {
+            /* these defaults are temporary while  */
+            executive: context.owner.address,
+            governor: context.owner.address,
+            guardian: context.owner.address,
+            ...contractRoles[contractKey],
+            /* contractRoleOverrides allow tests to pass in an address override to make role based testing more clear */
+            ...contractRoleOverrides[contractKey] as Partial<DefaultRoles>,
+        };
+        await contract.connect(signerAdmin).setRoles(roles.executive, roles.governor, roles.guardian);
+        const args: string[] = initArgs[contractKey] || [];
+        await contract.connect(signerAdmin).init.apply(null, [args]);
```

Comment:
Both init and setRoles now happen consistently on all contracts. All contracts are expected to have these methods. Args for init added in the initArgs object above.

In order to make the call signatures consistent and avoid typescript errors here, all init methods expect an array of addresses as the only argument. This is how all contract init methods already function (only addresses are provided to init). If this changes for some reason in the future and you want to pass additional arguments to init just change this line to
```
contract.connect(signerAdmin).init.apply(null, [args])
```
and then make add the per contract arguments to initArgs above like:
```
{
  ...
  exchanger: [[ address1, address2 ... ], someOtherArg],
  ...
}
```